gateway-js/src/loadSupergraphSdlFromStorage.ts

```diff
@@ -67,10 +120,68 @@ export async function loadSupergraphSdlFromStorage({
         'content-type': 'application/json',
       },
     });
+
+    const oobResponse = await result.json();
+    if (!oobResponse?.data?.reportError) {
+      throw new Error("An error occured while reporting your error to Apollo: "
+      + fetchErrorMsg + result.status + ' ' + result.statusText);
+    }
```

Comment:
Four things:

* `occured` missing another `r` 👉 `occurred`
* Maybe rather than "your error" we should be more precise?  In particular, it's not really _my_ error.  It's more like, "Seems like Apollo had an error and _additionally_, this program encountered an error while reporting Apollo's error to them.", right?
* This is not something that was introduced by this PR, but I find it odd that `fetchErrorMsg` is not literally _an error message that came from trying to `fetch` something_, but is instead _a generic message **prefix** that we staple on the front of other error messages_.  Further, I find it odd that we embed the trailing colon in `fetchErrorMsg` because it requires me to always go look and remember if it has the colon or not and it's used in concatenation a lot.  Like, what if I wanted to use the error without 

  i.e., it would be more intuitive to me as the developer to use this error message like:

  ```javascript
      `${errMsgPrefix}: ${result.status} ${result.statusText}`
  ```

  rather than 

  ```javascript
      `${fetchErrorMsg}${result.status} ${result.statusText}`
  ```

  Is just me? 😆 

* With the `fetchErrorMsg` included, this ends up as _really_ long error message. The existing error message will still propagate up, right? Do we want to print it again as an error of an error? e.g.,:

  ```
  An error occurred while reporting your error to Apollo: An error occurred while fetching your schema from Apollo: 418 I'm a teapot
  ```

As a straw-dog suggestion:
	  
```suggestion
    throw new Error(
      `Out-of-band error reporting failed: ${result.status} ${result.statusText}`);
  }
```